activerecord/test/cases/type/adapter_specific_registry_test.rb

```diff
@@ -82,8 +82,8 @@ class AdapterSpecificRegistryTest < ActiveRecord::TestCase
 
       assert_equal type.new, registry.lookup(:foo)
       assert_equal type.new(:ordered_arg), registry.lookup(:foo, :ordered_arg)
-      assert_equal type.new(keyword: :arg), registry.lookup(:foo, keyword: :arg)
-      assert_equal type.new(keyword: :arg), registry.lookup(:foo, keyword: :arg, adapter: :postgresql)
+      assert_equal type.new({ keyword: :arg }), registry.lookup(:foo, { keyword: :arg })
+      assert_equal type.new({ keyword: :arg }), registry.lookup(:foo, { keyword: :arg }, adapter: :postgresql)
```

Comment:
It actually addresses the warning but not sure if this is a good way to address it.

Reply:
Originally these tests `registry.lookup` accepts both `:ordered_arg` and `keyword: :arg`, now tests only for ordered arguments.
We should use a `type` which accepts both `:ordered_arg` and `keyword: :arg`.

Reply:
Thanks for the review. I have kept the test line 84 which takes `:ordered_arg` at line 84 as it is because it does not raise this warnings. Let me know if I understand something incorrectly.

Reply:
I meant that `registry.lookup(:foo, { keyword: :arg })` tests ordered hash argument `{ keyword: :arg }`, not keyword argument `keyword: :arg`.

Keep all assertions as it is.

```diff
diff --git a/activerecord/test/cases/arel/support/fake_record.rb b/activerecord/test/cases/arel/support/fake_record.rb
index 864f47e245..205bdeb41b 100644
--- a/activerecord/test/cases/arel/support/fake_record.rb
+++ b/activerecord/test/cases/arel/support/fake_record.rb
@@ -89,7 +89,7 @@ def quote(thing)
   end
 
   class ConnectionPool
-    class Spec < Struct.new(:config)
+    class Spec < Struct.new(:adapter, keyword_init: true)
     end
 
     attr_reader :spec, :connection
diff --git a/activerecord/test/cases/type/adapter_specific_registry_test.rb b/activerecord/test/cases/type/adapter_specific_registry_test.rb
index b58bdd5549..91bccde8cb 100644
--- a/activerecord/test/cases/type/adapter_specific_registry_test.rb
+++ b/activerecord/test/cases/type/adapter_specific_registry_test.rb
@@ -76,7 +76,11 @@ class AdapterSpecificRegistryTest < ActiveRecord::TestCase
     end
 
     test "construct args are passed to the type" do
-      type = Struct.new(:args)
+      type = Class.new do
+        def initialize(args)
+          @args = args
+        end
+      end
       registry = Type::AdapterSpecificRegistry.new
       registry.register(:foo, type)
 
@@ -117,7 +121,11 @@ class AdapterSpecificRegistryTest < ActiveRecord::TestCase
 
     test "registering adapter specific modifiers" do
       decoration = Struct.new(:value)
-      type = Struct.new(:args)
+      type = Class.new do
+        def initialize(args)
+          @args = args
+        end
+      end
       registry = Type::AdapterSpecificRegistry.new
       registry.register(:foo, type)
       registry.add_modifier({ array: true }, decoration, adapter: :postgresql)
```

Reply:
Sorry the above diff won't work.

Corrected here.

```diff
diff --git a/activerecord/test/cases/type/adapter_specific_registry_test.rb b/activerecord/test/cases/type/adapter_specific_registry_test.rb
index b58bdd5549..1093c4feca 100644
--- a/activerecord/test/cases/type/adapter_specific_registry_test.rb
+++ b/activerecord/test/cases/type/adapter_specific_registry_test.rb
@@ -76,7 +76,6 @@ class AdapterSpecificRegistryTest < ActiveRecord::TestCase
     end
 
     test "construct args are passed to the type" do
-      type = Struct.new(:args)
       registry = Type::AdapterSpecificRegistry.new
       registry.register(:foo, type)
 
@@ -117,7 +116,6 @@ class AdapterSpecificRegistryTest < ActiveRecord::TestCase
 
     test "registering adapter specific modifiers" do
       decoration = Struct.new(:value)
-      type = Struct.new(:args)
       registry = Type::AdapterSpecificRegistry.new
       registry.register(:foo, type)
       registry.add_modifier({ array: true }, decoration, adapter: :postgresql)
@@ -131,5 +129,17 @@ class AdapterSpecificRegistryTest < ActiveRecord::TestCase
         registry.lookup(:foo, array: true, adapter: :sqlite3)
       )
     end
+
+    TYPE = Class.new do
+      attr_reader :args
+
+      def initialize(args = nil)
+        @args = args
+      end
+
+      def ==(other) self.args == other.args end
+    end
+
+    private def type; TYPE end
   end
 end
```

Reply:
Thanks for the kind review and the code suggestion.